src/api.rs

```diff
@@ -1111,6 +1114,72 @@ impl AudioClient {
     pub fn get_timing_mode(&self) -> Option<TimingMode> {
         self.timingmode
     }
+
+    /// Get the Acoustic Echo Cancellation Control.
+    /// If it succeeds, the capture endpoint supports control of the loopback reference endpoint for AEC.
+    pub fn get_aec_control(&self) -> WasapiRes<AcousticEchoCancellationControl> {
+        let control = unsafe {
+            self.client
+                .GetService::<IAcousticEchoCancellationControl>()?
+        };
+        Ok(AcousticEchoCancellationControl { control })
+    }
+
+    /// Get the Audio Effects Manager.
+    pub fn get_audio_effects_manager(&self) -> WasapiRes<AudioEffectsManager> {
+        let manager = unsafe { self.client.GetService::<IAudioEffectsManager>()? };
+        Ok(AudioEffectsManager { manager })
+    }
+
+    /// Set the category of an audio stream.
+    pub fn set_audio_stream_category(&self, category: AUDIO_STREAM_CATEGORY) -> WasapiRes<()> {
+        let audio_client_2 = self.client.cast::<IAudioClient2>()?;
+
+        let audio_client_property = AudioClientProperties {
+            cbSize: size_of::<AudioClientProperties>() as u32,
+            eCategory: category,
+            ..Default::default()
+        };
+
+        unsafe { audio_client_2.SetClientProperties(&audio_client_property as *const _)? };
+        Ok(())
+    }
+
+    /// Check if the Acoustic Echo Cancellation (AEC) is supported.
+    pub fn is_aec_supported(&self) -> WasapiRes<bool> {
+        if !self.is_aec_effect_present()? {
+            return Ok(false);
+        }
+
+        match unsafe { self.client.GetService::<IAcousticEchoCancellationControl>() } {
+            Ok(_) => Ok(true),
+            Err(err) if err == E_NOINTERFACE.into() => Ok(false),
+            Err(err) => Err(err.into()),
+        }
+    }
+
+    fn is_aec_effect_present(&self) -> WasapiRes<bool> {
+        // IAudioEffectsManager requires Windows 11 (build 22000 or higher).
+        let audio_effects_manager = match self.get_audio_effects_manager() {
+            Ok(manager) => manager,
+            Err(WasapiError::Windows(win_err)) if win_err == E_NOINTERFACE.into() => {
+                // Audio effects manager is not supported, so clearly not present.
+                return Ok(false);
+            }
+            Err(err) => return Err(err),
+        };
+
+        if let Some(audio_effects) = audio_effects_manager.get_audio_effects()? {
+            for effect in audio_effects {
```

Comment:
This would look nicer with an iterator and any()
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any

Reply:
Done.